.github/workflows/ccpp.yml

```diff
@@ -3,27 +3,58 @@ name: CMake
 on: [push, pull_request]
 
 jobs:
-  build:
-    runs-on: ubuntu-latest
-    
+  build_and_test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        COMPILER: [gcc, clang]
+        LIBMBUS_SANITIZE_MEMORY: ["ON", "OFF"]
+
     steps:
+    - name: setup (linux)
+      if: startsWith(matrix.os, 'ubuntu') && startsWith(matrix.COMPILER, 'clang') 
+      run: |
+        sudo apt install clang
+
     - uses: actions/checkout@v2
     - name: build examples and tests
-      run: rm -rf build || true && mkdir build && cd build && cmake .. -DLIBMBUS_BUILD_EXAMPLES=ON -DLIBMBUS_BUILD_TESTS=ON -DLIBMBUS_ENABLE_COVERAGE=ON && cmake --build . -j && cd ..
+      run: |
+        if [[ ${{matrix.COMPILER}} == *"clang"* ]]; then
+          export CC=clang
+          export CXX=clang++
+        fi
+        rm -rf build || true
+        mkdir build
+        cd build
+        cmake .. \
+        -DLIBMBUS_BUILD_EXAMPLES=ON \
+        -DLIBMBUS_BUILD_TESTS=ON \
+        -DLIBMBUS_ENABLE_COVERAGE=ON \
+        -DLIBMBUS_SANITIZE_MEMORY=${{matrix.LIBMBUS_SANITIZE_MEMORY}}
+        cmake --build . -j
+        cd ..
 
     - name: generate test frames
       run: ./test/generate-xml.sh test/test-frames
 
     - name: install and run gcovr
+      if: startsWith(matrix.COMPILER, 'gcc') 
       run: sudo pip install gcovr && gcovr build/.
 
-  debian:
+  debian_package:
```

Comment:
This looks unrelated

Reply:
Yes, just random cleanup

Reply:
So please remove the cleanup from the commit